portafly/src/components/pages/product/CreateProductPage.tsx

```diff
@@ -0,0 +1,182 @@
+import React, {
+  useState, FormEventHandler, FocusEventHandler, useEffect
+} from 'react'
+
+import {
+  PageSection,
+  TextContent,
+  Text,
+  Card,
+  CardBody,
+  Form,
+  FormGroup,
+  TextInput,
+  TextArea,
+  ActionGroup,
+  Button
+} from '@patternfly/react-core'
+import { useTranslation } from 'i18n/useTranslation'
+import { useHistory, Redirect } from 'react-router'
+import { createProduct, NewProduct } from 'dal/products'
+import { useAsync } from 'react-async'
+import { useAlertsContext } from 'components/util'
+import { ValidationException } from 'utils'
+
+type Validations = Record<string, {
+  validation: 'default' | 'success' | 'error',
+  errors?: string[]
+}>
+
+const CreateProductPage = () => {
+  // @ts-ignore missing strings file
+  const { t } = useTranslation('createProduct')
+  const { goBack } = useHistory()
+  const { addAlert } = useAlertsContext()
+
+  const [name, setName] = useState('')
+  const [systemName, setSystemName] = useState('')
+  const [description, setDescription] = useState('')
+  const [validations, setValidations] = useState<Validations>({
+    name: { validation: 'default' },
+    system_name: { validation: 'default' },
+    description: { validation: 'default' }
+  })
+
+  const {
+    isPending, error, run, data
+  } = useAsync({ deferFn: createProduct })
+
+  useEffect(() => {
+    if (error) {
+      if (Object.prototype.hasOwnProperty.call(error, 'validationErrors')) {
+        const { validationErrors } = (error as unknown as ValidationException)
+        const newValidations: Validations = {}
+        Object.keys(validationErrors).forEach((id) => {
+          newValidations[id] = {
+            validation: 'error',
+            errors: validationErrors[id]
+          }
+        })
+        setValidations({ ...validations, ...newValidations })
+      } else {
+        addAlert({ id: String(Date.now()), title: error.message, variant: 'danger' })
+      }
+    }
```

Comment:
So `validationErrors` are not provided in the `data` object, but within `error` and not always. We might want to leverage this inside the http module. Then, would be easy to check that if `error`is true, data will be `validationErrors`. Since the parsing of this can't be done without modifying `useAsync`

Reply:
No, error can be anything, including a validation error we're gonna need to check for it either way. Consider as if `Object.prototype.hasOwnProperty.call(error, 'validationErrors')` were `typeof error === 'ValidationException'`...
There's no way to work around this. It's either check the error for `validationErrors` or the data, but I think it belongs to error.